pennylane_lightning/core/src/simulators/lightning_gpu/measurements/MeasurementsGPU.hpp

```diff
@@ -0,0 +1,762 @@
+
+        const PrecisionT mean_square =
+            innerProdC_CUDA(ob_sv.getData(), ob_sv.getData(), ob_sv.getLength(),
+                            device_id, stream_id, ob_sv.getCublasCaller())
+                .x;
+
+        const PrecisionT squared_mean = static_cast<PrecisionT>(std::pow(
+            innerProdC_CUDA(this->_statevector.getData(), ob_sv.getData(),
+                            this->_statevector.getLength(), device_id,
+                            stream_id, this->_statevector.getCublasCaller())
+                .x,
+            2));
+        return (mean_square - squared_mean);
+    };
+
+    /**
+     * @brief Variance of an observable.
+     *
+     * @param matrix Square matrix in row-major order.
+     * @param wires Wires where to apply the operator.
+     * @return Floating point with the variance of the observable.
+     */
+    auto var(const std::vector<ComplexT> &matrix,
+             const std::vector<size_t> &wires) -> PrecisionT {
+        StateVectorT ob_sv(this->_statevector.getData(),
+                           this->_statevector.getLength());
+        ob_sv.applyMatrix(matrix, wires);
+
+        auto device_id = ob_sv.getDataBuffer().getDevTag().getDeviceID();
+        auto stream_id = ob_sv.getDataBuffer().getDevTag().getStreamID();
+
+        const PrecisionT mean_square =
+            innerProdC_CUDA(ob_sv.getData(), ob_sv.getData(), ob_sv.getLength(),
+                            device_id, stream_id, ob_sv.getCublasCaller())
+                .x;
+
+        const PrecisionT squared_mean = static_cast<PrecisionT>(std::pow(
+            innerProdC_CUDA(this->_statevector.getData(), ob_sv.getData(),
+                            this->_statevector.getLength(), device_id,
+                            stream_id, this->_statevector.getCublasCaller())
+                .x,
+            2));
+        return (mean_square - squared_mean);
+    };
+
+    /**
+     * @brief Variance for a list of observables.
+     *
+     * @tparam op_type Operation type.
+     * @param operations_list List of operations to measure.
+     * Square matrix in row-major order or string with the operator name.
+     * @param wires_list List of wires where to apply the operators.
+     * @return Floating point std::vector with the variance of the
+     observables.
+     */
+    template <typename op_type>
+    auto var(const std::vector<op_type> &operations_list,
+             const std::vector<std::vector<size_t>> &wires_list)
+        -> std::vector<PrecisionT> {
+        PL_ABORT_IF(
+            (operations_list.size() != wires_list.size()),
+            "The lengths of the list of operations and wires do not match.");
+
+        std::vector<PrecisionT> expected_value_list;
+
+        for (size_t index = 0; index < operations_list.size(); index++) {
+            expected_value_list.emplace_back(
+                var(operations_list[index], wires_list[index]));
+        }
+
+        return expected_value_list;
```

Comment:
As above, I think we wanted test coverage here.

Reply:
hmmm, it's tested [here](https://github.com/PennyLaneAI/pennylane-lightning/blob/5f20309098d83616f989c5581b359a6eccf27a74/pennylane_lightning/core/src/simulators/lightning_gpu/measurements/tests/Test_StateVectorCudaManaged_Measure.cpp#L222-L225). It's strange that codecov didn't capture it.